Builder/FormContractor.php

```diff
@@ -84,31 +84,53 @@ public function getDefaultOptions($type, FieldDescriptionInterface $fieldDescrip
         $options = array();
         $options['sonata_field_description'] = $fieldDescription;
 
-        if ($type == 'sonata_type_model' || $type == 'sonata_type_model_list') {
-            if ($fieldDescription->getOption('edit') == 'list') {
+        // NEXT_MAJOR: Check only against FQCNs when dropping support for Symfony <2.8
+        if (in_array($type, array(
+            'sonata_type_model',
+            'Sonata\AdminBundle\Form\Type\ModelType',
+            'sonata_type_model_list',
+            'Sonata\AdminBundle\Form\Type\ModelTypeList',
+            'sonata_type_model_hidden',
+            'Sonata\AdminBundle\Form\Type\ModelHiddenType',
+            'sonata_type_model_autocomplete',
+            'Sonata\AdminBundle\Form\Type\ModelAutocompleteType',
+        ), true)) {
+            if ($fieldDescription->getOption('edit') === 'list') {
```

Comment:
No need to just vote :-1: because you don't like it. It's a convention so please follow it.